src/util/commit.js

```diff
@@ -115,5 +113,7 @@ exports.deserialize = (data, callback) => {
     }
   }
 
-  setImmediate(() => callback(null, res))
+  visibility.removeEnumerableProperties(res, ['gitType'])
```

Comment:
There are only 2 uses of `visibility` and they both have this one property, they (and visibility.js) could be replaced with `Object.defineProperty(res, 'gitType', { enumerable: false })`. I'll leave that call up to you though, I think you know my thoughts on the `defineProperty()` use and the re-use of visibility.js across the projects.